mods/mc_tutorial/init.lua

```diff
@@ -0,0 +1,1708 @@
+    end,
+    [mc_tutorial.ACTION.LOOK_YAW] = function(event)
+        return nil, "look at yaw (horizontal) "..(event.dir and math.deg(event.dir).."°" or "[?]")
+    end,
+    [mc_tutorial.ACTION.LOOK_PITCH] = function(event)
+        return nil, "look at pitch (vertical) "..(event.dir and math.deg(event.dir).."°" or "[?]")
+    end,
+    [mc_tutorial.ACTION.LOOK_DIR] = function(event)
+        local yaw, pitch = vect_to_yp(event.dir)
+        return nil, "look in direction "..(event.dir and "(yaw = "..yaw.."°, pitch = "..pitch.."°)" or "[?]")
+    end,
+    [mc_tutorial.ACTION.POS_ABS] = function(event)
+        return nil, "go to position "..(event.pos and "(x = "..event.pos.x..", y = "..event.pos.y..", z = "..event.pos.z..")" or "[?]")
+    end,
+    [mc_tutorial.ACTION.GROUP] = function(event)
+        if event.g_type == mc_tutorial.GROUP.START then
+            return "#CCFFFF", "GROUP "..(event.g_id or "[?]").." {"
+        else
+            return "#CCFFFF", "} END GROUP "..(event.g_id or "[?]")
+        end
+    end,
+}
+
+function mc_tutorial.show_record_fs(player)
+    local pname = player:get_player_name()
+	if mc_tutorial.check_privs(player, mc_tutorial.recorder_priv_table) then
+        -- Tutorial formspec for recording a tutorial
+        local context = get_context(pname)
+        mc_tutorial.record.temp[pname] = mc_tutorial.record.temp[pname] or mc_tutorial.get_temp_shell()
+        local temp = mc_tutorial.record.temp[pname]
+
+        -- Get all recorded events
+        if not context.events then
+            local events = {}
+
+            for i,event in ipairs(temp.sequence) do
+                if event.action then
+                    local col, event_string = event_action_map[event.action](event)
+                    table.insert(events, (col or "")..minetest.formspec_escape(event_string or ""))
+                else
+                    table.insert(events, "#FFCCCC"..minetest.formspec_escape("[?]"))
+                end
+            end
+            context.events = events
+        end
+
+        -- Get all available rewards
+        if not context.rewards then
+            local rewards = {}
+            local selected_rewards = {}
+            for priv,_ in pairs(minetest.registered_privileges) do
+                if mc_helpers.tableHas(temp.on_completion.privs, priv) then
+                    table.insert(selected_rewards, {col = "#FFCCFF", s = minetest.formspec_escape("[P] ")..priv})
+                else
+                    table.insert(rewards, {col = "#FFCCFF", s = minetest.formspec_escape("[P] ")..priv})
+                end
+            end
+
+            local item_map = {
+                ["tool"] = {col = "#CCFFFF", s_pre = minetest.formspec_escape("[T] ")},
+                ["node"] = {col = "#CCFFCC", s_pre = minetest.formspec_escape("[N] ")},
+            }
+            for item,def in pairs(minetest.registered_items) do
```

Comment:
Will add this check when the tutorial is giving tools so that rewards can still be added to tutorials and given if `mc_toolhandler` is later removed or deactivated, but they aren't given if `mc_toolhandler` is actively managing them